src/dra/5298C.c

```diff
@@ -2,7 +2,728 @@
+                func_800F0CD8(0);
+                if (g_StageId == STAGE_RTOP) {
+                    DestroyEntities(0x40);
+                    for (i = 0; i < 8; i++) {
+                        g_unkGraphicsStruct.D_8009742C[i] = 0;
+                    }
+                    D_80073074 = (s32)g_Camera.posX.i.hi;
+                    D_8007307C = (s32)g_Camera.posY.i.hi;
+                    g_api.o.InitRoomEntities(D_801375BC->objLayoutId);
+                    g_api.o.Update();
+                    func_800F0CD8(0);
+                    func_800F0CD8(0);
+                    DestroyEntities(0x40);
+                    for (i = 0; i < 8; i++) {
+                        g_unkGraphicsStruct.D_8009742C[i] = 0;
+                    }
+                }
+                D_801375A4 = D_8013759C - PLAYER.posX.val;
+                D_801375A8 = D_801375A0 - PLAYER.posY.val;
+                for (i = 0, ent = &g_Entities[0]; i < 256; i++, ent++) {
+                    if (ent->flags & 0x20000) {
+                        ent->posX.val -= D_801375A4;
+                        ent->posY.val -= D_801375A8;
+                    }
+                }
+                i = 0;
+                func_80121F14(-D_801375A4, -D_801375A8);
+                D_801375A4 = D_801375A6;
+                D_801375A8 = D_801375AA;
+                for (i = 0, prim = &g_PrimBuf[0]; i < 1024; i++, prim++) {
+                    if (prim->blendMode & 0x100) {
+                        switch (prim->type & 0xf) {
+                        case PRIM_G4:
+                        case PRIM_GT4:
+                            prim->x3 -= D_801375A4;
+                            prim->y3 -= D_801375A8;
+                        case PRIM_GT3:
+                            prim->x2 -= D_801375A4;
+                            prim->y2 -= D_801375A8;
+                        case PRIM_LINE_G2:
+                            prim->x1 -= D_801375A4;
+                            prim->y1 -= D_801375A8;
+                        case PRIM_TILE:
+                        case PRIM_SPRT:
+                            prim->x0 -= D_801375A4;
+                            prim->y0 -= D_801375A8;
+                        case PRIM_NONE:
+                        case PRIM_ENV:
+                            break;
+                        }
+                    }
+                }
+                if (g_StageId != STAGE_RTOP) {
+                    DestroyEntities(0x40);
+                    for (i = 0; i < 8; i++) {
+                        g_unkGraphicsStruct.D_8009742C[i] = 0;
+                    }
+                    D_80073074 = g_Camera.posX.i.hi;
+                    D_8007307C = g_Camera.posY.i.hi;
+                    g_api.o.InitRoomEntities(D_801375BC->objLayoutId);
+                    g_api.o.Update();
+                    func_800F0CD8(0);
+                    func_800F0CD8(0);
+                    DestroyEntities(0x40);
+                    for (i = 0; i < 8; i++) {
```

Comment:
i < LEN(D_8009742C)

Reply:
Done, thanks for the reminder, I always forget LEN() exists.